### DISCUSSION TITLE
How to enable cache for Curobo docker container

### DISCUSSION THREAD

**Discussion by emfebert** [NVlabs/curobo#131] [Software Q&A] [answered]
Hi,

I'm using the docker container curobo_docker:user_isaac_sim_2023.1.0 and the cache is disabled as indicated by the message in the top-right corner of the isaac-sim gui.

How can I enable the cache?
The loading times for the curobo examples are extremely slow (3-4 minutes) every time.

I'm using isaac-sim as a native gui in the docker.

Thanks!

**Comment by Sepirence** [comment]
Nvidia developer forum can be your answer.
https://forums.developer.nvidia.com/t/in-isaac-sim-docker-container-how-can-i-cache-on/255315/3
Cache button on top-right corner only works with omniverse isaac-sim.

I just used the simulator and this repository only 3 weeks, but if you are struggling with loading cache like rtx rendering, how about check that you mounted cache volume when you run the container first?

**Reply by emfebert** [reply]
The native Isaac sim cache works.

I followed the instructions for the issac sim container here:
https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_container.html
But the cache isn't working there either.

I'm definitely mounting the cache paths since I use their command:
docker run --name isaac-sim --entrypoint bash -it --gpus all -e "ACCEPT_EULA=Y" --rm --network=host
-e "PRIVACY_CONSENT=Y"
-v ~/docker/isaac-sim/cache/kit:/isaac-sim/kit/cache:rw
-v ~/docker/isaac-sim/cache/ov:/root/.cache/ov:rw
-v ~/docker/isaac-sim/cache/pip:/root/.cache/pip:rw
-v ~/docker/isaac-sim/cache/glcache:/root/.cache/nvidia/GLCache:rw
-v ~/docker/isaac-sim/cache/computecache:/root/.nv/ComputeCache:rw
-v ~/docker/isaac-sim/logs:/root/.nvidia-omniverse/logs:rw
-v ~/docker/isaac-sim/data:/root/.local/share/ov/data:rw
-v ~/docker/isaac-sim/documents:/root/Documents:rw
nvcr.io/nvidia/isaac-sim:2023.1.1

**Reply by Sepirence** [reply, marked as answer]
Follow this site then you may find answer.
https://docs.omniverse.nvidia.com/isaacsim/latest/installation/install_faq.html#isaac-sim-setup-set-omni-server

Default isaac-sim container sets download path to `"http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Isaac/2023.1.0"`

You can use your localhost omniverse cache server by adding arguments to your docker run.
```
-e "OMNI_SERVER=omniverse://localhost/NVIDIA/Assets/Isaac/2023.1.1" 
```

Remind that this cache is for storing asset, not for performance of simulation. You may allocate more resources to your container if simulation is too slow.
